### PR TITLE
[bug 406066] Improve usage of simple names for java.lang classes

### PR DESCRIPTION
**Pull request by kthoms**
When the import manager is created with a type that collides with java.lang, the flag thisCollidesWithJavaLang is set. But in method allowsSimpleName it is not considered that the actual type name that should be shortened is different from the one with that the ImportManager was created.
Signed-off-by: Karsten Thoms <karsten.thoms@itemis.de>

### CODE REVIEW

**Comment by oehme** [comment]
Can one of the admins verify this patch?

File: tests/org.eclipse.xtend.core.tests/src/org/eclipse/xtend/core/tests/compiler/CompilerBug406066Test.xtend [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+/**
+ * @author Karsten Thoms - Initial contribution and API
+ */
+class CompilerBug406066Test extends AbstractXtendCompilerTest {
+	
+	@Test
+	def testBug406066_01() {
+		assertCompilesTo('''
+			class Error {
+				def static err(Integer i) {
```
Comment (szarnekow) [inline tests/org.eclipse.xtend.core.tests/src/org/eclipse/xtend/core/tests/compiler/CompilerBug406066Test.xtend]: Could you add a usage of java.lang.Error in this method, too?

Comment (kthoms) [inline tests/org.eclipse.xtend.core.tests/src/org/eclipse/xtend/core/tests/compiler/CompilerBug406066Test.xtend]: Added 2 more test cases.

**Comment by szarnekow** [comment]
Jenkins test

**Comment by szarnekow** [comment]
Jenkins test

**Comment by szarnekow** [comment]
@kthoms unfortunately the patch breaks existing tests (which may have a bogus expectation, though). Could you double check these?

**Comment by kthoms** [comment]
Fixed test expectation

**Comment by szarnekow** [comment]
Thank you for the patch, Karsten.

**Pull request merged**